lib/plurimath/mathml/parse.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "parslet"
+module Plurimath
+  class Mathml
+    class Parse < Parslet::Parser
+      rule(:parse_record) do
+        parse_class.as(:class) |
+          parse_unicode_symbols.as(:symbol) |
+          parse_symbols.as(:symbol) |
+          match["a-zA-Z"].as(:text) |
+          match(/[0-9]/).repeat(1).as(:number) |
+          str("")
+      end
+
+      rule(:tag) { (parse_tag(:open) >> iteration.as(:iteration) >> parse_tag(:close)).as(:tag) | parse_text_tag.as(:tag) }
+
+      rule(:sequence) { (tag >> sequence.as(:sequence)) | tag }
+
+      rule(:iteration) { (sequence >> iteration.as(:iteration)) | parse_record }
+
+      rule(:expression) { parse_tag(:open) >> iteration.as(:iteration) >> parse_tag(:close) }
+
+      root :expression
+
+      def parse_class
+        Constants::CLASSES.reduce do |expr, tag|
+          expr = str(expr) if expr.is_a?(String)
+          expr | str(tag)
+        end
+      end
+
+      def parse_unicode_symbols
+        Constants::UNICODE_SYMBOLS.keys.reduce do |expr, tag|
+          expr = str(expr) if expr.is_a?(Symbol)
+          expr | str(tag)
+        end
+      end
+
+      def parse_symbols
+        Constants::SYMBOLS.keys.reduce do |expr, tag|
+          expr = str(expr) if expr.is_a?(Symbol)
+          expr | str(tag)
+        end
+      end
```

Comment:
These functions and the `tags` function on line 65 have similar code can we merge these into a common method?

Reply:
Agree